contracts/SponsorFaucet.sol

```diff
@@ -23,96 +23,176 @@ contract SponsorFaucet is Ownable, Pausable, ReentrancyGuard {
+                .gas_amount_accumulated
+                .add(gas_bound)
+                .sub(gas_balance);
+        } else {
+            //simply ignore the refund and dapp will get less than total_limit
+            dapps[dapp].gas_amount_accumulated = dapps[dapp]
+                .gas_amount_accumulated
+                .add(gas_bound);
         }
         emit applied(msg.sender, dapp, gas_bound);
     }
 
     function _applyForCollateral(address dapp) internal {
-        uint256 collateral_balance = internal_sponsor.getSponsoredBalanceForCollateral(dapp);
+        uint256 collateral_balance = internal_sponsor
+            .getSponsoredBalanceForCollateral(dapp);
         internal_sponsor.setSponsorForCollateral.value(collateral_bound)(dapp);
         address last_sponsor = internal_sponsor.getSponsorForCollateral(dapp);
-        if(last_sponsor == address(this) || last_sponsor == address(0x0000000000000000000000000000000000000000)) {
-            dapps[dapp].collateral_amount_accumulated = dapps[dapp].collateral_amount_accumulated.add(collateral_bound).sub(collateral_balance);
+        if (
+            last_sponsor == address(this) ||
+            last_sponsor == address(0x0000000000000000000000000000000000000000)
+        ) {
+            dapps[dapp].collateral_amount_accumulated = dapps[dapp]
+                .collateral_amount_accumulated
+                .add(collateral_bound)
+                .sub(collateral_balance);
+        } else {
+            //simply ignore the refund and dapp will get less than total_limit
+            dapps[dapp].collateral_amount_accumulated = dapps[dapp]
+                .collateral_amount_accumulated
+                .add(collateral_bound);
         }
         emit applied(msg.sender, dapp, collateral_bound);
     }
 
     //accept sponsor's cfx
-    function () external payable {
-    }
+    function() external payable {}
 
     //withdraw to specific address by amount
-    function withdraw(address payable sponsor, uint256 amount) public onlyOwner nonReentrant whenPaused {
+    function withdraw(address payable sponsor, uint256 amount)
+        public
+        onlyOwner
+        nonReentrant
+        whenPaused
+    {
         require(address(this).balance >= amount, "amount too high");
         (bool success, ) = sponsor.call.value(amount)("");
         require(success, "withdraw failed");
     }
 
     //set bounds for sponsorship
-    function setBounds(uint256 gasTotalLimit, uint256 collateralTotalLimit, uint256 gasBound, uint256 collateralBound, uint256 upperBound) public onlyOwner {
-        require(upperBound.mul(1000) <= gasBound, 'upperBound too high');
+    function setBounds(
+        uint256 gasTotalLimit,
+        uint256 collateralTotalLimit,
+        uint256 gasBound,
+        uint256 collateralBound,
+        uint256 upperBound
+    ) public onlyOwner {
+        require(upperBound.mul(1000) <= gasBound, "upperBound too high");
```

Comment:
better to avoid magic number or add necessary comments.

Reply:
fixed